handlers/handler-auth/src/main/java/io/servicecomb/authentication/consumer/RSACoumserTokenManager.java

```diff
@@ -0,0 +1,77 @@
+package io.servicecomb.authentication.consumer;
+
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SignatureException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Date;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import org.apache.commons.lang3.RandomStringUtils;
+
+import io.servicecomb.authentication.RSAAuthenticationToken;
+import io.servicecomb.foundation.common.utils.RSAUtils;
+import io.servicecomb.foundation.token.AuthenticationTokenManager;
+import io.servicecomb.foundation.token.RSAKeypair4Auth;
+import io.servicecomb.serviceregistry.RegistryUtils;
+
+public class RSACoumserTokenManager implements AuthenticationTokenManager {
+
+	private ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+	
+	private RSAAuthenticationToken token;
+
+	@Override
+	public String getToken() {
+		readWriteLock.readLock().lock();
+		if(null != token && valid(token.fromat()))
```

Comment:
token object already have generateTime
but in this place, we ignore it, and calculate it again, why?

Reply:
accept